src/HexagonOptimize.cpp

```diff
@@ -151,6 +152,7 @@ std::vector<Pattern> casts = {
     { "halide.hexagon.sxt.vh", i32(wild_i16x), Pattern::InterleaveResult },
 
     // Saturating narrowing casts
+    { "halide.hexagon.satub.shr.vh.h", u8c(wild_i16x/wild_i16), Pattern::DeinterleaveOperands | Pattern::IsOp1PowerOfTwo },
```

Comment:
Can you add a test to simd_op_check for this pattern? Also, are there other types that we should implement for this instruction?